lib/nodejs/lib/thrift/connection.js

```diff
@@ -81,15 +81,7 @@ var Connection = exports.Connection = function(stream, options) {
   });
 
   this.connection.addListener("error", function(err) {
-    // Only emit the error if no-one else is listening on the connection
-    // or if someone is listening on us
-    if (self.connection.listeners('error').length === 1 ||
-        self.listeners('error').length > 0) {
-      self.emit("error", err);
-    }
```

Comment:
This code doesn't make any sense; emitting `error` with no listeners is a no-op, so we might as well do it always.